examples/eks-msk-opensearch/terraform/templates/fluent-bit.yaml

```diff
@@ -0,0 +1,89 @@
+# Template out the image version and service account name to use so that they can be modified from Terraform.
+image:
+  tag: ${image_version}
+
+serviceAccount:
+  name: ${service_account_name}
+
+# Tolerate all taints so that all nodes are logged.
+tolerations:
+  - key: ""
+    operator: "Exists"
+    effect: "NoExecute"
+
+# Pipeline configuration.
+config:
+  # This is a config for the FluentBit application running the pipeline.
+  # We set the flush interval, log level and where our parsers are stored.
+  # We additionally expose the HTTP service so that Terraform can ping it.
+  service: |
+    [SERVICE]
+        Daemon Off
+        Flush 1
+        Log_Level info
+        Parsers_File parsers.conf
+        HTTP_Server On
+        HTTP_Listen 0.0.0.0
+        HTTP_Port 2020
+        Health_Check On
+  # We read all of the logs for output by docker's json file logging engine.
+  # Every line appended to a file that matches the Path field will generate a record in the pipeline.
+  inputs: |
+    [INPUT]
+        Name                tail
+        Tag                 kube.*
+        Path                /var/log/containers/*.log
+        # Parser              cri
+        Mem_Buf_Limit       5MB
+        Skip_Long_Lines     On
+  # Grep Filter drops logs that are only whitespace.
+  # Kubernetes Filter appends K8s information to all outgoing logs.
+  filters: |
+    # Dispose of logs that are empty
+    [FILTER]
+        Name                grep
+        Match               kube.*
+        Exclude             $message /^\s+$/
+
+    # Dispose of logs for liveness/readiness check
+    [FILTER]
+        Name                grep
+        Match               kube.*
+        Exclude             $message /.*GIN-debug*./
+        Exclude             $message /.*[GIN]*./
+
+    [FILTER]
+        Name                kubernetes
+        Match               kube.*
+        K8S-Logging.Parser  On
+        K8S-Logging.Exclude On
+
+    # Include logs for configured namespace
+    [FILTER]
+        Name                grep
+        Match               kube.*
+        Regex               $kubernetes['namespace_name'] ${namespaces}
```

Comment:
this will filter out all logs that do not contain the namespace name you set.

Another option would be to only collect logs for the configured namespace. This would be more performant than collecting all of them and then discarding the ones you do not want. 

You can do this in the tail pattern, the log file name contains the namespace name: https://stackoverflow.com/questions/47915287/where-are-kubernetes-pods-logfiles

In the container insights daemonset they do this: https://github.com/aws-samples/amazon-cloudwatch-container-insights/blob/master/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/fluent-bit/fluent-bit.yaml#L144

Though I think they are using the container name not the namespace name... since hte log file name is:

`/var/log/containers/<pod-name>_<namespace>_<container-name-container-id>.log -> /var/log/pods/<some-uuid>/<container-name>_0.log
`

I think a specifc namespace can be collected with:

```
Name                tail
        Tag                 kube.*
        Path                /var/log/containers/*${namespaces}*.log
```

I think that would work. Please try it.

Reply:
Hey- did you try this? What do you think?

Reply:
It seems like the goal here is to support multiple namespaces? and just set the kafka topic based on the namespace? If so, then I think we do not want to filter out logs not matching this single namespace?